marklogic-data-hub/src/test/java/com/marklogic/hub/impl/ClearUserModulesTest.java

```diff
@@ -0,0 +1,83 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.admin.QueryOptionsManager;
+import com.marklogic.client.admin.ResourceExtensionsManager;
+import com.marklogic.client.admin.ServerConfigurationManager;
+import com.marklogic.client.admin.TransformExtensionsManager;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.QueryOptionsListHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ClearUserModulesTest extends AbstractHubCoreTest {
+    ClearUserModulesTest() {
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        Assumptions.assumeTrue(this.isVersionCompatibleWith520Roles());
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        Path servicesFile = this.getHubProject().getModulesDir().resolve("services");
+        servicesFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules/services/mlBatches.sjs"), servicesFile.resolve("myResourceExtension.sjs").toFile());
+        Path optionsFile = this.getHubProject().getModulesDir().resolve("options");
+        optionsFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules-staging/options/default.xml"), optionsFile.resolve("myOptions.xml").toFile());
+        Path transformsFile = this.getHubProject().getModulesDir().resolve("transforms");
+        transformsFile.toFile().mkdirs();
+        FileUtils.copyInputStreamToFile(classLoader.getResourceAsStream("ml-modules/transforms/hubEntitySearchTransform.sjs"), transformsFile.resolve("myTransform.sjs").toFile());
+        this.installProjectInFolder("flow-runner-test");
+    }
+
+    @Test
+    void testClearUserModules() {
+        ServerConfigurationManager configMgr = this.runAsDataHubDeveloper().newFinalClient().newServerConfigManager();
+        QueryOptionsManager finalOptionsManager = configMgr.newQueryOptionsManager();
+        Map<String, String> optionsMap = ((QueryOptionsListHandle)finalOptionsManager.optionsList(new QueryOptionsListHandle())).getValuesMap();
+        Assertions.assertEquals(2, optionsMap.keySet().size());
+        Assertions.assertTrue(optionsMap.containsKey("myOptions"));
+        String query = "cts.estimate(cts.collectionQuery('hub-core-module'))";
+        int hubCoreModuleCount = Integer.parseInt((String)this.getHubClient().getModulesClient().newServerEval().javascript(query).evalAs(String.class));
```

Comment:
I think that before you install the project with user modules, you need to get counts of all docs in the modules database, all docs in hub-core-module, and then counts of each REST extension type that DHF installs - services, transforms, and options. 

Then you install the user modules project, and you verify that the counts are now increased by the appropriate amount.

Then you clear user modules, and you verify the counts are back to what they used to be.

To simplify all of this, I recommend creating a class - ModuleCounts - that has 5 fields, one for each count. Then make a private method that returns a ModuleCounts, doing all the queries necessary to get the correct amount.

I know that's a lot to test, but it's much better to automate this, this would be a real pain to test manually. And because of REST extensions resulting in multiple modules, we really need to verify individual counts of each extension type.